data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbClientWrapper.java

```diff
@@ -0,0 +1,174 @@
+
+    static final String ITEM_DOES_NOT_EXIST_EXPRESSION = "attribute_not_exists(sourcePartitionKey)";
+    static final String ITEM_EXISTS_AND_HAS_LATEST_VERSION = "attribute_exists(sourcePartitionKey) and version = :v";
+
+    private final DynamoDbEnhancedClient dynamoDbEnhancedClient;
+    private DynamoDbTable<DynamoDbSourcePartitionItem> table;
+
+
+    private DynamoDbClientWrapper(final String region, final String stsRoleArn) {
+        this.dynamoDbEnhancedClient = DynamoDbClientFactory.provideDynamoDbEnhancedClient(region, stsRoleArn);
+    }
+
+    public static DynamoDbClientWrapper create(final String region, final String stsRoleArn) {
+        return new DynamoDbClientWrapper(region, stsRoleArn);
+    }
+
+    public boolean tryCreateTable(final String tableName,
+                                  final ProvisionedThroughput provisionedThroughput) {
+        boolean createdTable = false;
+
+        this.table = dynamoDbEnhancedClient.table(tableName, TableSchema.fromBean(DynamoDbSourcePartitionItem.class));
+
+        try {
+            final CreateTableEnhancedRequest createTableEnhancedRequest = CreateTableEnhancedRequest.builder()
+                            .provisionedThroughput(provisionedThroughput)
+                            .build();
+            table.createTable(createTableEnhancedRequest);
+            createdTable = true;
+        } catch (final ResourceInUseException e) {
+            LOG.info("The table creation for {} was already triggered by another instance of data prepper", tableName);
+        }
+
+        try(final DynamoDbWaiter dynamoDbWaiter = DynamoDbWaiter.create()) {
+            final DescribeTableRequest describeTableRequest = DescribeTableRequest.builder().tableName(tableName).build();
+            final ResponseOrException<DescribeTableResponse> response = dynamoDbWaiter
+                    .waitUntilTableExists(describeTableRequest)
+                    .matched();
+
+            final DescribeTableResponse describeTableResponse = response.response().orElseThrow(
+                    () -> new RuntimeException(String.format("Table %s was not created successfully", tableName))
+            );
+
+            LOG.info("DynamoDB table {} was created successfully for source coordination", describeTableResponse.table().tableName());
+        }
+
+        return createdTable;
+    }
+
+    public boolean tryCreatePartitionItem(final DynamoDbSourcePartitionItem dynamoDbSourcePartitionItem) {
+        try {
+            table.putItem(PutItemEnhancedRequest.builder(DynamoDbSourcePartitionItem.class)
+                    .item(dynamoDbSourcePartitionItem)
+                    .conditionExpression(Expression.builder()
+                            .expression(ITEM_DOES_NOT_EXIST_EXPRESSION)
+                            .build())
+                    .build());
+            return true;
+        } catch (final ConditionalCheckFailedException e) {
+            return false;
```

Comment:
This boolean is just used to log at the moment (https://github.com/graytaylor0/data-prepper/blob/e79d030ea3cf2a659e7678f1a897b73fc4ebc45e/data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java#L115). 

For ProvisionedThroughput in this case, I guess we could catch it in the `DynamoDbSourceCoordinationStore` and retry this function up to a certain point. However, the ProvisionedThroughputException is only thrown after the exponential backoff and retry configured by the client.

With the current code hitting a ProvisionedThroughputException would mean the partition does not get picked up. However, it will be retried when the partition creation supplier gets  called when no partitions are available.